src/executable.js

```diff
@@ -0,0 +1,124 @@
+import { SnsWasmCanister } from '@dfinity/nns';
+import { Principal } from '@dfinity/principal';
+import { initSnsWrapper } from '@dfinity/sns';
+import fs from 'fs';
+
+const loadJSON = (path) =>
+  JSON.parse(fs.readFileSync(new URL(path, import.meta.url)));
+
+const ICP_SYMBOL = 'ICP';
+const MAINNET_SNS_WASM_CANISTER_ID = Principal.fromText(
+  'qaa6y-5yaaa-aaaaa-aaafa-cai'
+);
+
+const TokensJson = loadJSON('./tokenlist.json');
+
+const generateImage = async (base64String, path) => {
+  let base64Image = base64String.split(';base64,').pop();
+  fs.writeFile(
+    new URL(path, import.meta.url),
+    base64Image,
+    { encoding: 'base64', recursive: true },
+    function (err) {
+      console.log('File created');
+    }
+  );
+};
+
+const updateTokenListJson = async (data, path) => {
+  fs.writeFile(
+    new URL(path, import.meta.url),
+    JSON.stringify(data),
+    function (err) {
+      if (err) throw err;
+      console.log('complete');
+    }
+  );
+};
+
+class TokenList {
+  static async create() {
```

Comment:
function is a bit long, can we break it up a bit. Difficult to understand what it is doing without some comments.